requirements.txt

```diff
@@ -0,0 +1,7 @@
+cython>=0.26
+future>=0.15
+ipopt>=3.13.3
```

Comment:
I don't think we need to be this strict. I'm pretty sure it works with 3.11, 3.12, for example. We want to be able to build against versions that are packaged with ubuntu distros, brew, etc.

Reply:
Is this the case for Windows too now? If so I'll change to `ipopt>=3.11`.

Reply:
Not sure. We can merge as is and then do some testing with the older versions. Maybe we should add a line in the CI to test 3.11.

Reply:
Ubuntu 20.04 ships 3.11: https://packages.ubuntu.com/source/focal/coinor-ipopt, so I think we should try to support back to 3.11 at least.

Reply:
Okay, merging as is and then investigating further sounds good.

With less-than-or-equal-to version requirements conda will always use the most recent available versions (assuming no conflicts) so I can't immediate think of a way of cleverly testing older packages in the CI without creating multiple different conda environments. Maybe there's a tool for this. Probably the best approach initially is to manually test `ipopt=3.11` and `ipopt=3.12` and then change the minimum version requirement in a future PR. Then if we find a tool (I guess something similar to tox) that can randomise the dependency versions during CI we could use that.

Reply:
Not sure how feasible support back to 3.11 will be as there is no 3.11 on conda-forge for macOS/Linux https://anaconda.org/conda-forge/ipopt/files?version=3.11.1&page=2

Reply:
There is also no 3.12 on conda-forge for Windows. But CI passes on Appveyor with 3.11 and 3.13, and CI passes on Travis with 3.12 and 3.13. I've also tested 3.12 and 3.13 locally on macOS with success. I think this means we can use `ipopt>=3.11`. CI will still pass as conda will use 3.13.x on all OSs as it's available, then we just need to think more about how to continually check backwards compatibility with 3.11 and 3.12 if that needs supporting.

Reply:
I'm fine it we only test a single matrix entry in linux or windows with 3.11 on CI. We can test the rest with the latest version.

Reply:
Okay, I'll add that.